payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherActivity.kt

```diff
@@ -0,0 +1,216 @@
+package com.stripe.android.googlepaylauncher
+
+import android.content.Intent
+import android.content.pm.ActivityInfo
+import android.os.Build
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import com.google.android.gms.wallet.AutoResolveHelper
+import com.google.android.gms.wallet.PaymentData
+import com.google.android.gms.wallet.PaymentDataRequest
+import com.google.android.gms.wallet.PaymentsClient
+import com.stripe.android.model.PaymentMethodCreateParams
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.view.AuthActivityStarterHost
+import kotlinx.coroutines.launch
+import org.json.JSONObject
+
+/**
+ * [GooglePayLauncherActivity] is used to return the result of a Google Pay operation.
+ * The activity's result will be a [GooglePayLauncher.Result].
+ *
+ * [GooglePayLauncherActivity] will:
+ * 1. Fetch the [StripeIntent]
+ * 2. Verify that Google Pay is available
+ * 3. Create a [PaymentDataRequest](https://developers.google.com/pay/api/web/reference/request-objects#PaymentDataRequest)
+ * 4. Confirm the [StripeIntent] using the Google Pay token
+ *
+ * Use [GooglePayLauncherContract] to start [GooglePayLauncherActivity].
+ *
+ * See [Troubleshooting](https://developers.google.com/pay/api/android/support/troubleshooting)
+ * for a guide to troubleshooting Google Pay issues.
+ */
+internal class GooglePayLauncherActivity : AppCompatActivity() {
+    private val paymentsClient: PaymentsClient by lazy {
+        PaymentsClientFactory(this).create(args.config.environment)
+    }
+    private val viewModel: GooglePayLauncherViewModel by viewModels {
+        GooglePayLauncherViewModel.Factory(
+            application,
+            args
+        )
+    }
+
+    private lateinit var args: GooglePayLauncherContract.Args
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (Build.VERSION.SDK_INT != Build.VERSION_CODES.O) {
+            // In Oreo, Activities where `android:windowIsTranslucent=true` can't request
+            // orientation. See https://stackoverflow.com/a/50832408/11103900
+            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+        }
+
+        overridePendingTransition(0, 0)
+        setResult(
```

Comment:
is this expected? seems the extras might be overridden later, otherwise is it possible to refactor it into `finishWithResult`?

Reply:
Removed